lib/rmt/cli/products.rb

```diff
@@ -40,24 +40,28 @@ def disable(target)
   protected
 
   def change_product(target, set_enabled)
-    repo_id = Integer(target, 10) rescue nil
-    if repo_id
-      change_product_mirroring_by_id(repo_id, set_enabled)
+    product_id = Integer(target, 10) rescue nil
+    products = []
+    if product_id
+      product = Product.find(product_id)
+      products << product unless product.nil?
     else
       identifier, version, arch = target.split('/')
-      RMT::CLI::Repos.change_product_mirroring(set_enabled, identifier, version, arch)
+      conditions = { identifier: identifier, version: version }
+      conditions[:arch] = arch if arch
+      products = Product.where(conditions)
     end
-  end
-
-  def change_product_mirroring_by_id(id, mirroring_enabled)
-    conditions = { mirroring_enabled: !mirroring_enabled } # to only update the repos which need change
-    conditions[:enabled] = true if mirroring_enabled
 
-    repo_count = Product.find(id).change_repositories_mirroring!(conditions, mirroring_enabled)
+    repo_count = repository_service.change_mirroring_by_product!(set_enabled, products)
+    puts "#{repo_count} repo(s) successfully #{set_enabled ? 'enabled' : 'disabled'}."
+  rescue ActiveRecord::RecordNotFound
+    warn "Product by id \"#{product_id}\" not found."
```

Comment:
`RMT::CLI::Error` should be raised here, for consistency as well as for setting the exit code.
We've got some special handling for it in `RMT::CLI::Base`, which hides the stack trace and sets the exit code to a non-zero value, depending on the type of exception.